students/khtruong/lesson_05/mailroom_v3.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+from datetime import date
+
+"""This module runs the mailroom script.
+"""
+
+data_dict = {'Jeff Bezos': [3.65, 54.50],
+             'Mark Zuckerberg': [36.54, 1.25, 54.87],
+             'Paul Allen': [17.38],
+             'William Gates III': [25.55, 33.33, 78.14]
+             }
+
+
+def menu_selection(prompt, selection_dict):
+    """Return options for user to select from."""
+    while True:
+        userinput = input(prompt)
+        if selection_dict.get(userinput) is None:
+            print('\n{} is not a valid selection. '
+                  'Please try again!'.format(userinput))
+        else:
+            if selection_dict.get(userinput)() == 'Exit Menu':
+                break
+
+
+def exit_menu():
+    """Return 'Exit Menu'."""
+    return 'Exit Menu'
+
+
+def send_thankyou_email():
+    """Return a menu selection to send thank you email to donor."""
+    menu_selection(thankyou_prompt, thankyou_dict)
+
+
+def list_donors():
+    """Return all donors name."""
+    [print(donor) for donor in data_dict.keys()]
```

Comment:
this isn't really what list comprehensions are intended to be used for.

comprehensions have a return value, and using it as a shortcut to just print a value is wasteful.

here is an alternative way it could be used with prints:
https://stackoverflow.com/questions/37084246/printing-using-list-comprehension